### ISSUE TITLE
Make release failed for linux/ppc64le

### ISSUE THREAD

**Issue by Random-Liu** [kubernetes/kubernetes#30384]
`make release` always fails with the following failure in my desktop:
```
hyperkube build for ppc64le is disabled. Creating dummy text file instead.
# k8s.io/kubernetes/cmd/kubemark
/usr/local/go/pkg/tool/linux_amd64/link: running powerpc64le-linux-gnu-gcc failed: exit status 1
/tmp.k8s/go-link-183829930/go.o: In function `k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/apis/extensions.RunAsUserStrategy.CodecEncodeSelf':
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/apis/extensions/types.generated.go:12942:(.text+0x2010efc): relocation truncated to fit: R_PPC64_REL24 (stub) against `runtime.convT2E'
/tmp.k8s/go-link-183829930/go.o: In function `k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/apis/extensions.(*FSGroupStrategyOptions).CodecDecodeSelf':
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/apis/extensions/types.generated.go:13075:(.text+0x2011980): relocation truncated to fit: R_PPC64_REL24 (stub) against `runtime.convI2E'
/tmp.k8s/go-link-183829930/go.o: In function `k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/apis/extensions.FSGroupStrategyType.CodecEncodeSelf':
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/apis/extensions/types.generated.go:13195:(.text+0x2012294): relocation truncated to fit: R_PPC64_REL24 (stub) against `runtime.convT2E'
/tmp.k8s/go-link-183829930/go.o: In function `k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/apis/extensions.(*SupplementalGroupsStrategyOptions).CodecDecodeSelf':
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/apis/extensions/types.generated.go:13328:(.text+0x2012d18): relocation truncated to fit: R_PPC64_REL24 (stub) against `runtime.convI2E'
/tmp.k8s/go-link-183829930/go.o: In function `k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/apis/extensions.SupplementalGroupsStrategyType.CodecEncodeSelf':
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/apis/extensions/types.generated.go:13448:(.text+0x201362c): relocation truncated to fit: R_PPC64_REL24 (stub) against `runtime.convT2E'
/tmp.k8s/go-link-183829930/go.o: In function `k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/apis/extensions.(*PodSecurityPolicyList).CodecDecodeSelf':
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/apis/extensions/types.generated.go:13640:(.text+0x2014558): relocation truncated to fit: R_PPC64_REL24 (stub) against `runtime.convI2E'
/tmp.k8s/go-link-183829930/go.o: In function `k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/apis/extensions.(*PodSecurityPolicyList).codecDecodeSelfFromArray':
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/apis/extensions/types.generated.go:13777:(.text+0x20151a0): relocation truncated to fit: R_PPC64_REL24 (stub) against `runtime.writebarrierptr'
/tmp.k8s/go-link-183829930/go.o: In function `k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/apis/extensions.(*NetworkPolicy).CodecDecodeSelf':
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/apis/extensions/types.generated.go:13963:(.text+0x2015e98): relocation truncated to fit: R_PPC64_REL24 (stub) against `runtime.convI2E'
/tmp.k8s/go-link-183829930/go.o: In function `k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/apis/extensions.(*NetworkPolicy).codecDecodeSelfFromMap':
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/apis/extensions/types.generated.go:14008:(.text+0x201623c): relocation truncated to fit: R_PPC64_REL24 (stub) against `runtime.writebarrierptr'
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/apis/extensions/types.generated.go:14014:(.text+0x20164d0): relocation truncated to fit: R_PPC64_REL24 (stub) against `runtime.writebarrierptr'
/tmp.k8s/go-link-183829930/go.o: In function `k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/apis/extensions.(*NetworkPolicy).codecDecodeSelfFromArray':
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/apis/extensions/types.generated.go:14094:(.text+0x20169a4): additional relocation overflows omitted from the output
collect2: error: ld returned 1 exit status
```

This happens to me and @coufon.

**Comment by david-mcmahon** [comment]
@luxas releases are on hold with this.  Can you take a look?

**Comment by david-mcmahon** [comment]
@spxtr @ixdy Is `make release` not used by any jenkins job?

**Comment by spxtr** [comment]
Whoops. #28575 made it do a quick-release even on post-submit Jenkins. @fejta

**Comment by david-mcmahon** [comment]
So if there's a need to expedite specific jobs with `make quick-release`, that seems reasonable, but we absolutely need a `make release` somewhere, even if we catch it post-submit.

**Comment by spxtr** [comment]
Agreed, I think that was a mistake.

**Comment by luxas** [comment]
cc @laboger @pensu

@ixdy you should disable kubemark builds for ppc64le in the same manner hyperkube has been disabled. The tricky thing here is that @laboger has fixed the issue in the golang linker, but it didn't make v1.7 unfortunately. So either we wait for v1.8 or use a self patched-golang builder (which is not ideal)

The post-submit tests should always block the SQ when make release fails, which was what you promised when we switched to quick-release in presubmit. I hope that was a mistake, but it should be fixed asap

**Comment by spxtr** [comment]
> The post-submit tests should always block the SQ when make release fails, which was what you promised when we switched to quick-release in presubmit. I hope that was a mistake, but it should be fixed asap

It was, fix is in #30640

**Comment by ixdy** [comment]
I'm looking into the kubemark compilation issue right now. It looks somewhat nontrivial to disable it for linux/ppc64le.

**Comment by ixdy** [comment]
Ah, I understand the `hyperkube` hack better now. I think I have something for `kubemark`.

**Comment by ixdy** [comment]
bad news: breakage is not limited to kubemark:
```
hyperkube and kubemark builds for ppc64le are disabled. Creating dummy text file instead.
hyperkube and kubemark builds for ppc64le are disabled. Creating dummy text file instead.
# k8s.io/kubernetes/cmd/kube-apiserver
k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/api.PersistentVolumeReclaimPolicy.CodecEncodeSelf: relocation for runtime.convT2E+220 is t
oo big: -33558660
k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/api.(*PersistentVolumeStatus).CodecDecodeSelf: relocation for runtime.convI2E+592 is too b
ig: -33559048
k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/api.(*PersistentVolumeList).CodecDecodeSelf: relocation for runtime.convI2E+592 is too big
: -33565120
k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/api.(*PersistentVolumeClaim).CodecDecodeSelf: relocation for runtime.convI2E+592 is too bi
g: -33571896
k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/api.(*PersistentVolumeClaim).codecDecodeSelfFromMap: relocation for runtime.writebarrierpt
r+860 is too big: -33558476
k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/api.(*PersistentVolumeClaim).codecDecodeSelfFromMap: relocation for runtime.writebarrierpt
r+1716 is too big: -33559332
k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/api.(*PersistentVolumeClaim).codecDecodeSelfFromArray: relocation for runtime.writebarrier
ptr+1692 is too big: -33561044
k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/api.(*PersistentVolumeClaim).codecDecodeSelfFromArray: relocation for runtime.writebarrier
ptr+1848 is too big: -33561200
k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/api.(*PersistentVolumeClaimList).CodecDecodeSelf: relocation for runtime.convI2E+592 is to
o big: -33578944
k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/api.(*PersistentVolumeClaimList).codecDecodeSelfFromMap: relocation for runtime.writebarri
erptr+860 is too big: -33565524
k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/api.(*PersistentVolumeClaimList).codecDecodeSelfFromMap: relocation for runtime.writebarri
erptr+1716 is too big: -33566380
k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/api.(*PersistentVolumeClaimList).codecDecodeSelfFromArray: relocation for runtime.writebar
rierptr+1592 is too big: -33567992
k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/api.(*PersistentVolumeClaimList).codecDecodeSelfFromArray: relocation for runtime.writebar
rierptr+1748 is too big: -33568148
k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/api.(*PersistentVolumeClaimSpec).CodecDecodeSelf: relocation for runtime.convI2E+592 is to
o big: -33585848
k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/api.(*PersistentVolumeClaimSpec).codecDecodeSelfFromMap: relocation for runtime.newobject+
796 is too big: -33575244
k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/api.(*PersistentVolumeClaimSpec).codecDecodeSelfFromMap: relocation for runtime.writebarri
erptr+1072 is too big: -33572640
k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/api.(*PersistentVolumeClaimSpec).codecDecodeSelfFromMap: relocation for runtime.writebarrierptr+1524 is too big: -33573092
k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/api.(*PersistentVolumeClaimSpec).codecDecodeSelfFromArray: relocation for runtime.writebarrierptr+1180 is too big: -33574484
k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/api.(*PersistentVolumeClaimSpec).codecDecodeSelfFromArray: relocation for runtime.newobject+1356 is too big: -33577540
k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/api.(*PersistentVolumeClaimSpec).codecDecodeSelfFromArray: relocation for runtime.writebarrierptr+1632 is too big: -33574936
k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/api.(*PersistentVolumeClaimStatus).CodecDecodeSelf: relocation for runtime.convI2E+592 is too big: -33591976
/usr/local/go/pkg/tool/linux_amd64/link: too many errors
# k8s.io/kubernetes/cmd/kube-controller-manager
k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/api/v1.(*PodSpec).String: relocation for runtime.mapiterinit+216 is too big: -33568568
k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/api/v1.(*PodSpec).String: relocation for runtime.mapiternext+412 is too big: -33567748
k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/api/v1.(*PodSpec).String: relocation for runtime.mapaccess1_faststr+620 is too big: -33558844
k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/api/v1.(*PodSpec).String: relocation for runtime.convT2E+2212 is too big: -33554468
k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/api/v1.(*PodSpec).String: relocation for runtime.convT2E+2428 is too big: -33554684
k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/api/v1.(*PodSpec).String: relocation for runtime.convT2E+2644 is too big: -33554900
k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/api/v1.(*PodSpec).String: relocation for runtime.convT2E+2860 is too big: -33555116
k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/api/v1.(*PodSpec).String: relocation for runtime.convT2E+3076 is too big: -33555332
k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/api/v1.(*PodSpec).String: relocation for runtime.convT2E+3292 is too big: -33555548
k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/api/v1.(*PodSpec).String: relocation for runtime.convT2E+3508 is too big: -33555764
k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/api/v1.(*PodSpec).String: relocation for runtime.convT2E+3956 is too big: -33556212
k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/api/v1.(*PodSpec).String: relocation for runtime.convT2E+4312 is too big: -33556568
k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/api/v1.(*PodSpec).String: relocation for runtime.convT2E+4528 is too big: -33556784
k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/api/v1.(*PodStatus).String: relocation for runtime.convT2E+176 is too big: -33559992
k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/api/v1.(*PodStatus).String: relocation for runtime.convT2E+392 is too big: -33560208
k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/api/v1.(*PodStatus).String: relocation for runtime.convT2E+748 is too big: -33560564
k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/api/v1.(*PodStatus).String: relocation for runtime.convT2E+964 is too big: -33560780
k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/api/v1.(*PodStatus).String: relocation for runtime.convT2E+1180 is too big: -33560996
k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/api/v1.(*PodStatus).String: relocation for runtime.convT2E+1396 is too big: -33561212
k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/api/v1.(*PodStatus).String: relocation for runtime.convT2E+1844 is too big: -33561660
k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/pkg/api/v1.(*PodTemplateList).String: relocation for runtime.convT2E+392 is too big: -33565616
/usr/local/go/pkg/tool/linux_amd64/link: too many errors
```

**Comment by svanharmelen** [comment]
PR #29740 was referenced as the PR that caused the build to break, but after testing it turned out that is not correct.

Not sure if it is of any value for this kind of issue, but it seems the commit that is actually causing the build to break is a PR merged at least two days (or more) before PR #29740 was merged, as reverting to 6aaa5292 and doing a `./build/release.sh` still gives me the same error.

As I'm currently on holiday I cannot spend more time (to find the exact commit and investigate it) on this without getting in trouble with my family :wink: but hope this maybe helps...

**Comment by ixdy** [comment]
PPC builds are disabled by default now. closing this issue in favor of #25886.
